### PR TITLE
Add support for BPMN groups and adapt subprocess visualization

### PR DESCRIPTION
**Pull request by matthiaslehnertum**
This PR intends to add support for groups within BPMN diagrams and adapts how subprocesses are rendered to comply with the BPMN 2.0 standard.

### Checklist
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes
- [x] I translated all the newly inserted strings into German and English

### Motivation and Context
Groups allow for easier entity management in larger BPMN diagrams - therefore this change adds support for them.
Additionally, to comply with the specification of the BPMN 2.0 standard, the way how subprocesses are rendered is updated.

### Steps for Testing
1. Add a group (box with dashed outline) to the canvas
2. Drop other elements inside the group
3. Trying dragging the group around to assure everything inside the group is moved as well
4. Delete group to ensure contained elements are deleted as well
5. Drag subprocess onto the canvas and make sure it is displayed as expected

### Test Coverage

| File | % Stmts | % Branch | % Funcs | % Lines |
|------|-------:|-----:|-----:|-----:|
 main/packages/bpmn/bpmn-group                                          |   94.73 |      100 |   66.66 |   94.73 |                                                
  bpmn-group-component.tsx                                              |     100 |      100 |     100 |     100 |                                                    
  bpmn-group.ts                                                         |   85.71 |      100 |      50 |   85.71 |
 main/packages/bpmn/bpmn-subprocess                                     |   95.45 |      100 |   66.66 |   95.45 |                          
  bpmn-subprocess-component.tsx                                         |     100 |      100 |     100 |     100 |                              
  bpmn-subprocess.ts                                                    |      80 |      100 |      50 |      80 |

### Screenshots
Screenshot 2023-10-26 at 11 53 31

### CODE REVIEW

File: src/tests/unit/packages/bpmn/bpmn-group/bpmn-group-component-test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { wrappedRender } from '../../../test-utils/render';
+import * as React from 'react';
+import { CSSProperties } from 'react';
+import { Multiline } from '../../../../../main/utils/svg/multiline';
+import { BPMNGroup } from '../../../../../main/packages/bpmn/bpmn-group/bpmn-group';
+import { BPMNGroupComponent } from '../../../../../main/packages/bpmn/bpmn-group/bpmn-group-component';
+
+// override getStringWidth, because it uses by jsdom unsupported SVGElement methods
```
Comment (loreanvictor) [inline src/tests/unit/packages/bpmn/bpmn-group/bpmn-group-component-test.tsx:8]: @matthiaslehnertum I don't need this patching for getting the test to pass. As with the other PR, perhaps this is a JSDOM versioning issue? I test using a fresh clone so perhaps I'm getting the latest compatible version.

**Review by loreanvictor — DISMISSED** [review]
Similar with the pool PR, I couldn't find a way to disassociate elements from each other. I can drag elements into a group and associate them but can't drag it out and have the element not be associated with that group. Note that dragging an element into another group does change the association. Additionally, dragging a group outside of its parent group will disassociate the two groups, which is in line with other elements in Apollon.

File: src/tests/unit/packages/bpmn/bpmn-group/bpmn-group-component-test.tsx (hunk 1 above)
Comment (matthiaslehnertum) [reply src/tests/unit/packages/bpmn/bpmn-group/bpmn-group-component-test.tsx]: @loreanvictor Thank you for the in-depth testing! :) I managed to get the disassociation issue fix. The SVG method patch also does not seem to be necessary still - so I removed that as well.

**Review by AlexanderG2207 — APPROVED** [review]
Tested locally with Google Chrome. Adding and removing elements from a group works well. Deleting a group and all of its children also works as expected. Finally, exporting the diagram to SVG with a BPMN group also works. All tests pass locally as well. Well done!

**Pull request merged**